GLMakie/src/drawing_primitives.jl

```diff
@@ -269,7 +270,7 @@ function draw_atomic(screen::Screen, scene::Scene, @nospecialize(x::Lines))
             data[:pattern] = ls
         else
             linewidth = gl_attributes[:thickness]
-            data[:pattern] = ls .* (to_value(linewidth) * 0.25)
+            data[:pattern] = ls * _mean(to_value(linewidth))
```

Comment:
I took this as somewhat of a shortcut for now, because `linestyle` needs some cleanup anyway.

Example with variable linewidth:
```julia
GLMakie.closeall()
widths = [57.37044620677474, 27.847264587380923, 56.663525479541775, 47.302310563997985, 56.36164168328704, 43.014590186214285, 58.832167079498525, 27.208798997411307, 17.12575358453462, 45.95171757911098, 26.771344098755126, 35.85649823581235, 53.08232474276624, 27.217846025535966, 43.98443255191129, 37.389819316502695, 18.07005596531466, 59.569472267387056, 50.919270815270814, 36.74914882840677, 53.08834081120422, 45.44455789568474, 40.147375157549135, 22.166337026901253, 50.53203686576642, 15.238811525483367, 12.780634172815944, 12.980713770063876, 28.175981476856755, 37.542112019663485]
ys = [0.3672275443344566, 0.13769475910008522, 0.22445059716611304, 0.5883310148108587, 0.7893622240334566, 0.3129809823712394, 0.953606981656488, 0.2770419071274035, 0.3048558242752848, 0.6782285082882588, 0.5447836889282002, 0.19905515156197973, 0.19770277191975638, 0.537590947032379, 0.28027667604204765, 0.8287252657638743, 0.9336781225747905, 0.1371941303349402, 0.5973028266375866, 0.3561025405662749, 0.43500848119792546, 0.36410392908891054, 0.9117308440489152, 0.8008057243055526, 0.2349037873519133, 0.21294938231466431, 0.8334978136186971, 0.5097696792342837, 0.21488349759400982, 0.3434422610208452]
fig = Figure()
ax = Axis(fig[1, 1])
# p = linesegments!(ax, ys, linewidth = widths./10, linecap = 4, linecap_length = widths./10, color = :red)
# p = linesegments!(ax, ys, linewidth = widths./10, linecap = 4, linecap_length = widths./10, linestyle = :dot)
p = lines!(ax, ys, linewidth = widths./10, linecap = 0, linecap_length = widths, color = :red)
p = lines!(ax, ys, linewidth = widths./10, linecap = 0, linecap_length = widths, linestyle = :dash)
screen = display(GLMakie.Screen(), fig)
```

produces
![Screenshot from 2022-12-31 17-07-54](https://user-images.githubusercontent.com/10947937/210149408-fc8e028d-fe2e-430d-b6e7-61c0729f543e.png)
![Screenshot from 2022-12-31 17-08-04](https://user-images.githubusercontent.com/10947937/210149412-a4dfc193-ff44-4af1-bae4-ce8427287049.png)

and pixel perfect results with constant linewidth (tested explicitly for linewidth = 50 on a straight line with lines and linesegments looks the same).

There is however some artifacting with `:dash` that got more frequent but isn't new. From master:

![Screenshot from 2022-12-31 17-19-36](https://user-images.githubusercontent.com/10947937/210149503-dc5b454c-cf53-4430-9043-302903e6fd8b.png)

Reply:
Oh right, I also changed the scaling with linewidth in `lines!` (not here but in the shader) which makes `:dash` shorter notably at large linestyles. It's now 3lw on, 3lw off like the pattern requests. master has weird scaling, maybe quadratic? For linewidth 50, it seems to be about 40.1lw on, 40.1lw off. For linewidth 10 about 10.5lw on 10.5lw off
```julia
lines([0, 0], linewidth = 50, linestyle = :dash)
```
![Screenshot from 2022-12-31 17-26-52](https://user-images.githubusercontent.com/10947937/210149681-f759aba6-fe52-4b86-96bc-0b28a56a65c7.png)